pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -311,16 +311,18 @@ func (s *DiscoveryServer) periodicRefreshMetrics() {
 // Push is called to push changes on config updates using ADS. This is set in DiscoveryService.Push,
 // to avoid direct dependencies.
 func (s *DiscoveryServer) Push(full bool, edsUpdates map[string]*EndpointShardsByService) {
-	if !full {
-		adsLog.Infof("XDS Incremental Push EDS:%d", len(edsUpdates))
-		go s.AdsPushAll(versionInfo(), s.globalPushContext(), false, edsUpdates)
-		return
-	}
 	// Reset the status during the push.
 	pc := s.globalPushContext()
 	if pc != nil {
 		pc.OnConfigChange()
 	}
+
+	if !full {
+		adsLog.Infof("XDS Incremental Push EDS:%d", len(edsUpdates))
+		go s.AdsPushAll(versionInfo(), s.globalPushContext(), false, edsUpdates)
+		return
+	}
+
```

Comment:
Why this change? and what is it doing? this whole code is super hacky :(

Reply:
Just move push status reset in front of  incremental eds.